BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -612,6 +609,48 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
         }
     }
 
+    /// Check if master password complies with organization policies
+    ///
+    /// - Parameters:
+    ///  - masterPassword: The user's master password
+    ///  - isPreAuth: Whether this flow is before or after authentication
+    ///  - username: The user's email address.
+    ///  - masterPasswordPolicy: The master password policies that the org has active.
+    ///
+    private func checkMasterPasswordPolicies(
+        _ masterPassword: String,
+        _ isPreAuth: Bool,
+        _ username: String,
+        _ masterPasswordPolicy: MasterPasswordPolicyResponseModel?
+    ) async throws {
+        var policy: MasterPasswordPolicyOptions?
+        if let model = masterPasswordPolicy {
+            policy = MasterPasswordPolicyOptions(
+                minComplexity: model.minComplexity ?? 0,
+                minLength: model.minLength ?? 0,
+                requireUpper: model.requireUpper ?? false,
+                requireLower: model.requireLower ?? false,
+                requireNumbers: model.requireNumbers ?? false,
+                requireSpecial: model.requireSpecial ?? false,
+                enforceOnLogin: model.enforceOnLogin ?? false
+            )
+        }
```

Comment:
🎨 One thing to improve this a bit is to have an `init?(responseModel: MasterPasswordPolicyResponseModel)` in `MasterPasswordPolicyOptions+Extensions` which would take care of building the options or returning `nil`.
```swift
init?(responseModel: MasterPasswordPolicyResponseModel) {
    guard let responseModel else { return nil }
    self.init(
        minComplexity: responseModel.minComplexity ?? 0,
        minLength: responseModel.minLength ?? 0,
        requireUpper: responseModel.requireUpper ?? false,
        requireLower: responseModel.requireLower ?? false,
        requireNumbers: responseModel.requireNumbers ?? false,
        requireSpecial: responseModel.requireSpecial ?? false,
        enforceOnLogin: responseModel.enforceOnLogin ?? false
    )
}
```
So then you can simplify this to:
```swift
let policy = MasterPasswordPolicyOptions(responseModel: masterPasswordPolicy)
```